Articles/ShortArticles/TheDeveloperCertificateOfOrigin.md

```diff
@@ -0,0 +1,147 @@
+
+Some major projects are using the DCO with Git commit sign-offs for handling contributions including the Linux Kernel<sup>[8]</sup> and more recently the source code for GitLab itself <sup>[3],[4]</sup>.
+In addition, the Linux Foundation's CII Best Practices Badge Program<sup>[9]</sup> contains the Silver-level item:
+
+* The project SHOULD have a legal mechanism where all developers of non-trivial amounts of project software assert that they are legally authorized to make these contributions.
+The most common and easily-implemented approach for doing this is by using a Developer Certificate of Origin (DCO)<sup>[7]</sup>, where users add "signed-off-by" in their commits and the project links to the DCO website.
+However, this MAY be implemented as a Contributor License Agreement (CLA), or other legal mechanism. {Met URL} <sup>[10]</sup>
+
+A few issues must be considered when adopting the DCO for accepting contributions.
+First, since anyone can use any committer name and email address they want with a Git commit (e.g. using `git commit --amend --author "any-author"` and therefore impersonate someone else), greater assurance that the person signing off on the DCO is actually the author of the commit can be added by requiring commits to be GPG signed using the `-S` option with `git commit -s -S`<sup>[11]</sup>.
+(However, requiring GPG signing of all commits adds a lot of extra overhead and complexity which may be too much for many developers and therefore more limited usages of GPG signing are possible to provide some elevated assurances <sup>[12]</sup>.
+It is unclear if any major project requires GPG signing of all commits to assert the DCO.)
+
+Second, for the DCO to be effective, every commit that contains a nontrivial contribution needs to contain the `Signed-off-by` line.
+Tools are available to assert that all commits in a GitHub Pull Request (PR) contain the `Signed-off-by` line<sup>[13]</sup>.
+(Note that tools to automate the signing of CLAs for GitHub PRs also exist if one wants to automate CLA signoffs as well<sup>[14]</sup>.)
+
+In summary, using a Developer Certificate of Origin (DCO) is a lightweight legal approach to provide cover in accepting contributions but it comes at the cost of requiring every nontrivial commit from a Pull Request for a proposed contribution to be signed-off by an outside contributor (and the process to go back and fix unsigned commits can be quite cumbersome for some people with poor Git knowledge and skills, or depending on how those commits and branches were constructed).
+Therefore, for some developer communities, the upstream technical overhead and mechanics of applying the DCO may be more of an impediment to accepting outside contributions than the barriers to getting outside contributors or organizations to sign CLAs.
+
+
```

Comment:
Here I'd like to add

### Disclaimer

This is not legal advice. Consult your own lawyer before any action that may have legal consequences.  The article is based on U.S. copyright law; other countries may differ.

Reply:
Should be addressed in new commit badd685f8f5bf9f11a7a345e495083b25ecd127e